src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1078 @@
+   *     from the failing assertion. For example: {@code 1) test10(ErrorTestLangMinimized)
+   *     java.lang.AssertionError: Contract failed: compareTo-equals on fraction1 and fraction4}
+   */
+  private static String normalizeJUnitOutput(String input) {
+    Scanner scn = new Scanner(input);
+
+    // JUnit output starts with index 1 for first failure
+    int index = 1;
+    // String to represent the result that we return
+    String res = "";
+    // Continue until we finish processing the input String
+    while (scn.hasNextLine()) {
+      String line = scn.nextLine();
+      // Look for a String of the form "i)" where i is the current failing
+      // test index.
+      if (line.startsWith(index + ") ")) {
+        res += line + "\n" + scn.nextLine() + "\n";
+        index++;
+      }
+    }
+    scn.close();
+    return res;
+  }
+
+  /**
+   * Write a compilation unit to a Java file. The file that is written to is the file path obtained
+   * by appending the suffix to the end of the file's name, before the file type suffix.
+   *
+   * @param compUnit the compilation unit to write to file
+   * @param filePath the absolute path to the input Java file
+   * @param suffix the suffix to append to the name of the new Java file
+   * @return {@code String} representing the absolute path to the newly written Java file. Returns
+   *     {@code null} if error occurred in writing to the new file
+   */
+  private static String writeToFile(CompilationUnit compUnit, String filePath, String suffix) {
+    // Rename the overall class to [old class name][suffix]
+    new ClassRenamer().visit(compUnit, new String[] {getClassName(filePath), suffix});
+
+    // Create a new string to represent the new file name
+    String newFileName =
+        new StringBuilder(filePath).insert(filePath.lastIndexOf('.'), suffix).toString();
+
+    // Write the compilation unit to the new file
+    try (BufferedWriter bw = new BufferedWriter(new FileWriter(newFileName))) {
+      bw.write(compUnit.toString());
+    } catch (IOException e) {
+      System.err.println("Error writing to file: " + newFileName);
+      e.printStackTrace();
+      System.exit(1);
+    }
+
+    return newFileName;
+  }
+
+  /**
+   * Given the absolute file path to a Java file, return the simple class name of the file.
+   *
+   * @param filePath absolute path to a Java file
+   * @return the simple class name
+   */
+  public static String getClassName(String filePath) {
+    if (filePath == null) {
+      return null;
+    }
+
+    // Get the name of the file without full path
+    String fileName = Paths.get(filePath).getFileName().toString();
+    // Remove .java extension from file name
+    return fileName.substring(0, fileName.indexOf('.'));
+  }
+
+  /**
+   * Rename the overall class. Visit every class and interface declaration, of which we expect there
+   * to be one in the input test suite. The overall class will be renamed to class name + suffix.
```

Comment:
If you added this in the user-visible minimizer documentation, that's OK.  It would be good to detect when it is violated (or perhaps handle that case in the future).